src/dynarray.c

```diff
@@ -0,0 +1,73 @@
+#include "dynarray.h"
+
+#include "specific/init.h"
+
+#include <stdlib.h>
+#include <string.h>
+
+struct DYNARRAY {
+    char *data;
+    size_t elem_size;
+    size_t used;
+    size_t size;
+};
+
+DYNARRAY *DynArray_Create(size_t elem_size)
```

Comment:
Not having the option to start at a decent size will cause lots of unnecessary reallocs when used for some structures.

Reply:
Like I mentioned in a previous comment I'm not sure I even want this, the code that uses this looks pretty dense. WDYT about making `RoomsToDraw`, the only struct that currently uses it, a static array of `MAX_ROOMS`?

Reply:
The interface is a bit heavy on the eyes, yes :) I like the idea of unlimited rooms, items and everything, but I don't like it making debugging harder for now. So I'd just make some big arrays for now and keep it for the future if there's a want/need.

Reply:
Changed back, and made sure we're not running into buffer overflows.